<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <arg name="world" default="wallWorld" />
  
  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(find wk2Assignment_3)/worlds/$(arg world).world"/>
  
    <arg name="use_sim_time" value="true"/>
    <arg name="debug" default="false"/>
    <arg name="gui" default="true"/>
    <arg name="headless" default="false"/>
    <arg name="paused" default="false"/>
  </include>
  
  <arg name="x" default="9"/>
  <arg name="y" default="1"/>
  <arg name="z" default="0"/>
  <arg name="roll" default="0"/>
  <arg name="pitch" default="0"/>
  <arg name="yaw" default="3.14"/>
  <!--use a small script called spawn_model to make a service call request to the gazebo_ros ROS node-->


  <param name="robot_description" command="$(find xacro)/xacro.py '$(find wk2Assignment_3)/urdf/wk2Bot3.xacro'" />
  <node name="spawn_urdf" pkg="gazebo_ros" type="spawn_model" output="screen"
     args="-urdf -model wk2Bot3 -param robot_description -x $(arg x) -y $(arg y) -z $(arg z) -Y $(arg yaw)" />

     
  <arg name="des_x" default="-4"/>
  <arg name="des_y" default="-4"/>
  <param name="des_pos_x" value="$(arg des_x)"/>
  <param name="des_pos_y" value="$(arg des_y)"/>
  <param name="init_x" value="$(arg x)"/>
  <param name="init_y" value="$(arg y)"/>
  <param name="th_yaw" value="2"/>
  <param name="th_dist" value="0.3"/> 
 
  <node pkg="wk2Assignment_3" type="GoToPoint.py" name="go_to_point" output="screen"/>
  <node pkg="wk2Assignment_3" type="FollowWall.py" name="follow_wall" output="screen"/>
  <node pkg="wk2Assignment_3" type="Bug0.py" name="bug0" output="screen"/>
  <node pkg="wk2Assignment_3" type="Bug1.py" name="bug1" output="screen"/>
  <node pkg="wk2Assignment_3" type="Bug2.py" name="bug2" output="screen"/>
  <node pkg="wk2Assignment_3" type="Client.py" name="client" output="screen"/>
  

</launch>